AdyenActions/Components/3DS2/Action handlers/ThreeDS2CoreActionHandler.swift

```diff
@@ -8,6 +8,20 @@ import Adyen
 import Adyen3DS2
 import Foundation
 
+// TODO: Remove this once 3ds2 sdk is released.
+extension NSError {
+    func base64Representation() -> String {
+        "base64Representation"
+    }
+}
+
+internal enum ThreeDS2CoreActionHandlerError: Error {
+    case cancellationAction(ThreeDSResult)
+    case underlyingError(Error)
+    case unknown(UnknownError)
+    case missingTransaction
```

Comment:
whats the reason we have two enum's `ThreeDS2CoreActionHandlerError` and `ThreeDS2ActionHandlerError`, why not just one, i.e just `ThreeDS2ActionHandlerError`, they seem to be identical

Reply:
![image](https://github.com/Adyen/adyen-ios/assets/96450350/5ce0c5c1-61cf-42eb-858f-e3736e57eac2)

I've kept an `Error` type at each layer. 

At the CoreAction level, we seem to create a `ThreeDSResult` type, 
but at the `Classic/Compact-Action` level we create the `ThreeDSActionHandlerResult` type.

Which is why we have different error types reflecting that.

Reply:
`ThreeDSActionHandlerResult` is fundamentally different than `ThreeDSResult`, but both `ThreeDS2ActionHandlerError` and `ThreeDS2CoreActionHandlerError` look identical :-)

Reply:
As we discussed. We will remove the error handling as we will not introduce a cancellationError as it would be a breaking change.